### PR TITLE
Refactor Tracked references endpoints

### PR DESCRIPTION
**Pull request by Migaroez**
### Prerequisites

- [ ] I have added steps to test this contribution in the description below

### Description
Tracked references are a subset of relations that show whether a certain item is required (depended on  by) something else. There for
- Tracked references have been moved under document and media cycles
- Endpoints have been renamed towards their purpose
- Unneeded data has been removed
- Models have been restructured

This will also allow us to further optimize the queries for each specific use-case later on

### Testing
Create some relations with dependencies by picking  content/media on other content/media. Then check whether the endpoints return the expected data.

### CODE REVIEW

File: src/Umbraco.Cms.Api.Management/Controllers/Document/References/AreReferencedDocumentController.cs [hunk 1]
```diff
@@ -28,16 +28,16 @@ public ItemsTrackedReferenceController(ITrackedReferencesService trackedReferenc
     ///     Used when bulk deleting content/media and bulk unpublishing content (delete and unpublish on List view).
     ///     This is basically finding children of relations.
     /// </remarks>
-    [HttpGet("item")]
+    [HttpGet("are-referenced")]
```
Comment (kjac) [inline src/Umbraco.Cms.Api.Management/Controllers/Document/References/AreReferencedDocumentController.cs:31]: I think `referenced-by` would be a more meaningful route?

File: src/Umbraco.Cms.Api.Management/Controllers/Document/References/ReferencesDocumentController.cs [hunk 2]
```diff
@@ -28,21 +28,20 @@ public ByIdTrackedReferenceController(ITrackedReferencesService trackedReference
     ///     Used by info tabs on content, media etc. and for the delete and unpublish of single items.
     ///     This is basically finding parents of relations.
     /// </remarks>
-    [HttpGet("{id:guid}")]
+    [HttpGet("{id:guid}/references")]
```
Comment (kjac) [inline src/Umbraco.Cms.Api.Management/Controllers/Document/References/ReferencesDocumentController.cs:31]: `{id}/referenced-by` would be good here - more explicit. I first thought this endpoint returned any items referenced by `id` - but it's the other way around, right?

File: src/Umbraco.Web.UI.Client
Comment (kjac) [inline src/Umbraco.Web.UI.Client:1]: Did you mean to update the submodule, or was this an unintentional change?

File: src/Umbraco.Infrastructure/Mapping/RelationModelMapDefinition.cs [hunk 3]
```diff
@@ -9,6 +9,7 @@ public class RelationModelMapDefinition : IMapDefinition
     public void DefineMaps(IUmbracoMapper mapper)
     {
         mapper.Define<RelationItemDto, RelationItemModel>((source, context) => new RelationItemModel(), Map);
+        mapper.Define<RelationItemDto, RelationItemModel>((source, context) => new RelationItemModel(), Map);
```
Comment (kjac) [inline src/Umbraco.Infrastructure/Mapping/RelationModelMapDefinition.cs:12]: Duplicate line?

File: src/Umbraco.Cms.Api.Management/Controllers/Document/References/AreReferencedDocumentController.cs (hunk 1 above)
Comment (Migaroez) [reply src/Umbraco.Cms.Api.Management/Controllers/Document/References/AreReferencedDocumentController.cs:31]: This endpoint returns which of the items you pass in are used by other items in a dependent way. So whether they are a reference on, aka are referenced by, another item

File: src/Umbraco.Cms.Api.Management/Controllers/Document/References/ReferencesDocumentController.cs (hunk 2 above)
Comment (Migaroez) [reply src/Umbraco.Cms.Api.Management/Controllers/Document/References/ReferencesDocumentController.cs:31]: This endpoint should return a list of all items it references, so for a document all the media/content/... that have been picked

**Review by kjac — APPROVED** [review]
Looks good 💪

**Pull request merged**